scripts/packages/VSCodeServer/src/repl.jl

```diff
@@ -59,6 +59,56 @@ function hideprompt(f)
     r
 end
 
+function hook_repl(repl)
+    if !isdefined(repl, :interface)
+        repl.interface = REPL.setup_interface(repl)
+    end
+    main_mode = get_main_mode(repl)
+
+    # TODO: set up REPL module ?
+    main_mode.on_done = REPL.respond(repl, main_mode; pass_empty = false) do line
+        quote
+            $(evalrepl)(Main, $line, $repl, $main_mode)
+        end
+    end
+end
+
+function evalrepl(m, line, repl, main_mode)
+    return try
+        JSONRPC.send_notification(conn_endpoint[], "repl/starteval", nothing)
+        try
+            repleval(m, line, REPL.repl_filename(repl, main_mode.hist))
+        catch err
+            display_repl_error(stderr, err, stacktrace(catch_backtrace()))
+            nothing
+        end
+    catch err
+        # This is for internal errors only.
+        Base.display_error(stderr, err, catch_backtrace())
+        nothing
+    finally
+        JSONRPC.send_notification(conn_endpoint[], "repl/finisheval", nothing)
+    end
+end
+
+function display_parse_error(io, err::Meta.ParseError)
+  printstyled(io, "ERROR: "; bold = true, color = Base.error_color())
+  printstyled(io, "syntax: ", err.msg; color = Base.error_color())
+  println(io)
+end
+
+# don't inline this so we can find it in the stacktrace
+@noinline repleval(m, code, file) = include_string(m, code, file)
```

Comment:
Why `include_string` over [`eval`](https://github.com/JunoLab/Atom.jl/blob/master/src/repl.jl#L196)?

Reply:
So we don't have to parse in an extra step. Not sure if I like it, but we can iterate on this in a follow-up.

Reply:
another benefit is we can specify `file` within `include_string` so that stacktrace shows `REPL[exec_cnt]` as a location, without fixing `Core.LineNumber`s in `Expr`.
And I don't see the downside of using it ? Not too sure about that though.

Reply:
The only slightly annoying thing about this is that everything's wrapped in a `LoadError`:
![image](https://user-images.githubusercontent.com/6735977/84376598-aad07180-abe1-11ea-9dea-951470bccc95.png)

Reply:
Fixed by https://github.com/julia-vscode/julia-vscode/pull/1331.